So the good news is that the pull image problem doesn't seem to be there anymore - huge thanks to @Xynnn007. We have some other issues though:
- On the s390x runner:
```
=== RUN   TestLibvirtSealedSecret
time="2026-07-22T16:31:48Z" level=info msg="set key resource: default/signing-key/sealed-secret"
    libvirt_test.go:159: Set signing key Secret failed with: exit status 2
--- FAIL: TestLibvirtSealedSecret (0.05s)
=== RUN   TestLibvirtKbsKeyRelease
time="2026-07-22T16:31:48Z" level=info msg="set key resource: caa/workload_key/test_key.bin"
    libvirt_test.go:187: SetSecret failed with: exit status 2
--- FAIL: TestLibvirtKbsKeyRelease (0.00s)
```
so it looks like we might have any issue with the new trustee?
- AWS is timing out on the cluster create
- x86 libvirt is timing out on the cluster create
It looks like these have failures pulling packages from ubuntu, and canonical has a partial outage, so we might have to wait to re-try them.

--- a/docs/Release-Process.md
+++ b/docs/Release-Process.md
@@ -15,8 +15,7 @@ The flow of releases should roughly be:
    version is picked) and [trustee releases](https://github.com/confidential-containers/trustee/releases).
     - This triggers [kata-containers](https://github.com/kata-containers/kata-containers) to update to these new versions in
       [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml) under
-      `externals.coco-guest-components.version`, `externals.coco-trustee` and the `image-rs` crate in the agent's
-      [`Cargo.toml`](https://github.com/kata-containers/kata-containers/blob/main/src/agent/Cargo.toml).
+      `externals.coco-guest-components.version` and `externals.coco-trustee`.
     - At this point it makes sense for us to stay in sync, by updating the guest-components and kbs that we use in peer pods,
       by changing the `oci.guest-components.reference` and `git.kbs.tag`values in [versions.yaml](../src/cloud-api-adaptor/versions.yaml).
       We should also bump the kata agent to the latest commit hash by updating `oci.kata-containers.reference` in

--- a/src/cloud-api-adaptor/go.mod
+++ b/src/cloud-api-adaptor/go.mod
@@ -19,7 +19,7 @@ require (
 	github.com/coreos/go-iptables v0.6.0
 	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/google/uuid v1.6.0
-	github.com/kata-containers/kata-containers/src/runtime v0.0.0-20260519201048-cec98e0d976b
+	github.com/kata-containers/kata-containers/src/runtime v0.0.0-20260720141120-cf82bb35c803
 	github.com/opencontainers/runtime-spec v1.2.1
 	github.com/stretchr/testify v1.11.1
 	github.com/vishvananda/netlink v1.3.1

--- a/src/cloud-api-adaptor/install/charts/peerpods/Chart.lock
+++ b/src/cloud-api-adaptor/install/charts/peerpods/Chart.lock
@@ -1,0 +1,12 @@
+dependencies:
+- name: kata-deploy
+  repository: oci://ghcr.io/kata-containers/kata-deploy-charts
+  version: 4.0.0
+- name: peerpodctrl
+  repository: file://../../../../peerpod-ctrl/chart
+  version: 0.3.1
+- name: peerpods-webhook
+  repository: file://../../../../webhook/chart
+  version: 0.3.1
+digest: sha256:21f90168320c110da2a761d9091b80a6c3c2647ec1743073651eccaf811afe19
+generated: "2026-07-21T10:33:48.36132+01:00"

--- a/src/cloud-api-adaptor/install/charts/peerpods/Chart.yaml
+++ b/src/cloud-api-adaptor/install/charts/peerpods/Chart.yaml
@@ -21,7 +21,7 @@ maintainers:
 
 dependencies:
   - name: kata-deploy
-    version: "0.0.0-dev"
+    version: "4.0.0"
     repository: "oci://ghcr.io/kata-containers/kata-deploy-charts"
     condition: kata-deploy.enabled
   - name: peerpodctrl

